server/v2/appmanager/service.go

```diff
@@ -13,7 +14,12 @@ import (
 )
 
 type AppManagerBuilder[T transaction.Tx] struct {
-	InitGenesis map[string]func(ctx context.Context, moduleGenesisBytes []byte) error
+	InitGenesis    map[string]func(ctx context.Context, moduleGenesisBytes []byte) error
+	PrepareHandler appmanager.PrepareHandler[T]
+}
+
+func (a *AppManagerBuilder[T]) Register(moduleName string, prepareFunc appmanager.PrepareHandler[T]) {
+	a.PrepareHandler = prepareFunc
 }
```

Comment:
`moduleName` isn't used here.

Reply:
right now do we expose preparehandler to modules or only at the top level? this meant to be top level only, in app.go this would be set

Reply:
ATM, we allows apps to implement and set a single PrepareProposal handler. However, in this handler, an app can have access to anything it wants.

e.g.

```go
type MyDummyPrepareHandler {
  ak AccountKeeper
  bk BankKeeper
  // ...
}
```

Reply:
the builder is removed in another pr, removed this function here for now. Does that solve the concern?